src/rpc/net.cpp

```diff
@@ -272,6 +272,49 @@ static RPCHelpMan getpeerinfo()
     };
 }
 
+static RPCHelpMan addwhitelist()
+{
+    return RPCHelpMan{"addwhitelist",
+                "\nAdd permission flags to the peers using the given IP address (e.g. 1.2.3.4) or\n"
+                "CIDR-notated network (e.g. 1.2.3.0/24). Uses the same permissions as -whitebind.\n"
+                "Additional flags in and out control whether permissions apply to incoming connections and/or outgoing (default: incoming only).\n",
+                {
+                    {"permissions", RPCArg::Type::ARR, RPCArg::Optional::NO, "An array of permission flags",
+                        {
+                            {"permission", RPCArg::Type::STR, RPCArg::Optional::OMITTED, ""}
+                        }
+                    }
+                },
+                RPCResult{RPCResult::Type::NONE, "", ""},
+                RPCExamples{
+                    HelpExampleCli("addwhitelist", "[\"noban@192.168.0.6:8333]\"")
+            + HelpExampleRpc("addwhitelist", "[\"noban@192.168.0.6:8333]\"")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    CConnman& connman = EnsureConnman(node);
+
+    const auto permissions{request.params[0].get_array()};
+
+    for (size_t i = 0; i < permissions.size(); i++) {
+        NetWhitelistPermissions subnet;
```

Comment:
nit: `m_subnet` is a member of `NetWhitelistPermissions`, isn't `permissions` a more accurate name?
```suggestion
        NetWhitelistPermissions permissions;
```

Reply:
Agreed.